api/src/opentelemetry/trace/span_id.cc

```diff
@@ -0,0 +1,29 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <opentelemetry/trace/span_id.h>
+
+#include <cstdint>
+#include <cstring>
+#include <string>
+
+namespace opentelemetry {
+namespace trace {
+
+SpanId::SpanId(const uint8_t *buf) { memcpy(rep_, buf, kSize); }
```

Comment:
Consider plain copy for perf:

```suggestion
SpanId::SpanId(const uint8_t *buf) { *this = *reinterpret_cast<const SpanId*>(buf); }
```

Reply:
In my experience, memcpy() usually compiles to a `MOV` on x86.

Reply:
This is an implementation detail. Let's please leave this as-is for now, and someone can come and speed it up later when we have tests and benchmarks etc. ;)